judge/views/contests.py

```diff
@@ -317,8 +317,12 @@ def get_context_data(self, **kwargs):
             raise Http404()
 
         dates = Contest.objects.aggregate(min=Min('start_time'), max=Max('end_time'))
-        min_month = dates['min'].year, dates['min'].month
-        max_month = max((dates['max'].year, dates['max'].month), (self.today.year, self.today.month))
+        min_month = self.today.year, self.today.month
+        max_month = self.today.year, self.today.month
+        if dates['min'].year != None:
```

Comment:
I thought it's `dates['min']` that is supposed to be `None`? Wouldn't this be `TypeError: NoneType has no attribute "year"`?

Also, do not do `!= None`, use `is None` when checking for `None`.